staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/interface.go

```diff
@@ -0,0 +1,75 @@
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fairqueuing
+
+import (
+	"context"
+	"time"
+)
+
+// QueueSetFactory is used to create QueueSet objects.
+type QueueSetFactory interface {
+	NewQueueSet(name string, concurrencyLimit, numQueues, queueLengthLimit, handSize int, requestWaitLimit time.Duration) (QueueSet, error)
+}
+
+// QueueSet is the abstraction for the queuing and dispatching
+// functionality of one non-exempt priority level.  It covers the
+// functionality described in the "Assignment to a Queue", "Queuing",
+// and "Dispatching" sections of
+// https://github.com/kubernetes/enhancements/blob/master/keps/sig-api-machinery/20190228-priority-and-fairness.md
+// .  Some day we may have connections between priority levels, but
+// today is not that day.
+type QueueSet interface {
+	// SetConfiguration updates the configuration
+	SetConfiguration(concurrencyLimit, desiredNumQueues, queueLengthLimit, handSize int, requestWaitLimit time.Duration) error
+
+	// A quiescing QueueSet drains as normal but does not admit any
+	// new requests. Passing a non-nil handler means the system should
+	// become quiescent, a nil handler means the system should become
+	//  non-quiescent. A call to Wait while the system is quiescent
+	// will be rebuffed by returning `quiescent=true`.  If all the
+	// queues have no requests  waiting nor executing while the system
+	// is quiescent then the handler will eventually be called with no
+	// locks held (even if the system becomes non-quiescent between the
+	// triggering state and the required call).
+	//
+	// The filter uses this for a priority level that has become
+	// undesired, setting a handler that will cause the priority level
+	// to eventually be removed from the filter if the filter still
+	// wants that.  If the filter later changes its mind and wants to
+	// preserve the priority level then the filter can use this to
+	// cancel the handler registration.
+	Quiesce(EmptyHandler)
+
+	// Wait uses the given hashValue as the source of entropy
+	// as it shuffle-shards a request into a queue and waits for
+	// a decision on what to do with that request.  If quiescent==true
+	//  at return then the QueueSet has become undesirable and the client
+	// should try to find a different QueueSet to use; execute and
+	// afterExecution are irrelevant in this case.  Otherwise, if execute
+	//  then the client should execute  the request and then call
+	// afterExecution().  Otherwise the client should not execute the
+	//  request and afterExecution is irrelevant.
+	Wait(ctx context.Context, hashValue uint64) (quiescent, execute bool, afterExecution func())
+}
+
+// EmptyHandler is used to notify callers when all the queues
```

Comment:
The phrase "notify callers" here is confusing.  There is only one call in this interface, and it notifies the _callee_.

Reply:
Updated to:
```
// EmptyHandler is used to notify the callee when all the queues
// of a QueueSet have been drained.
```